core/io/net_socket.h

```diff
@@ -67,6 +67,9 @@ class NetSocket : public Reference {
 
 	virtual bool is_open() const = 0;
 	virtual int get_available_bytes() const = 0;
+	virtual Error get_address(IP_Address &local_address, uint16_t &local_port) const = 0;
+	virtual Error get_address(IP_Address &local_address) const = 0;
+	virtual Error get_address(uint16_t &local_port) const = 0;
```

Comment:
probably worth joining in a single method

Reply:
And renamed to `getsockname` for consistency

Reply:
In my understanding, joining in a single method will require to change the parameters to pointers (so they can be optional), and then call the method with NULLs on the parameters we are not interested in.
Is this the intended result of your suggestion, or is there another approach?

Renaming to `getsockname` seems to go against all the other methods naming convention and it will overlap with the method from `sys/socket.h` (I personally try to avoid this). May I suggest to rename it to `get_socket_address` or `get_socket_name`?

Reply:
Joined into a single method and renamed to `get_socket_address`.